pkg/network/event_common.go

```diff
@@ -285,11 +285,15 @@ type ConnectionStats struct {
 	ProtocolStack   protocols.Stack
 
 	// keep these fields last because they are 1 byte each and otherwise inflate the struct size due to alignment
-	Direction        ConnectionDirection
-	SPortIsEphemeral EphemeralPortType
-	IntraHost        bool
-	IsAssured        bool
-	IsClosed         bool
+	Direction ConnectionDirection
+	IntraHost bool
+	IsAssured bool
+	IsClosed  bool
+}
+
+// SPortIsEphemeral returns whether the source port is in the ephemeral range
+func (c *ConnectionStats) SPortIsEphemeral() EphemeralPortType {
+	return IsPortInEphemeralRange(c.Family, c.Type, c.SPort)
```

Comment:
Why not caching the result, and by that guaranteeing it will be a one-time lazy load?

Reply:
You mean caching it outside of ConnectionStats ?